can_read_write.py

```diff
@@ -0,0 +1,94 @@
+# This scipt must run on Pi startup
+
+import can
+import time
+import os
+import sys
+import json
+from threading import Thread
+
+# info handled; specified by https://docs.google.com/document/d/1kUU54jQZAB9nwCM-iA96Kj0fXJ6RNw9nAcBffzji5cU/edit
+# full list of PIDs at https://en.wikipedia.org/wiki/OBD-II_PIDs
+PIDS = {'VEHICLE_CHARGE': 0x2F,
+        'VEHICLE_SPEED': 0x0D,
+}
+
+
+PID_REQUEST = 0x7DF
+PID_REPLY = 0x7E8
+
+OUTPUT_PATH = os.path.join(sys.path[0], 'data.json')
+
+
+class FilteredBufferReader(can.BufferedReader):
+    def __init__(self):
+        can.BufferedReader.__init__(self)
+
+    def on_message_received(self, msg):
+        if msg.arbitration_id == PID_REPLY and msg.data[2] in PIDS.keys():
+            self.buffer.put(msg)
+
+
+def request_task():
```

Comment:
I would call this `request_data()`. Calling it `request_task()` makes it sound like this is requesting a task, which is obviously not right. That also implies that it's supposed to be called from a background thread, which doesn't have to be the case.